server/src/main/java/org/apache/cloudstack/acl/RoleManagerImpl.java

```diff
@@ -146,10 +150,89 @@ public RoleVO doInTransaction(TransactionStatus status) {
         });
     }
 
+    @Override
+    @ActionEvent(eventType = EventTypes.EVENT_ROLE_CREATE, eventDescription = "creating Role")
+    public Role createRole(String name, Role role, String description) {
+        checkCallerAccess();
+        return Transaction.execute(new TransactionCallback<RoleVO>() {
+            @Override
+            public RoleVO doInTransaction(TransactionStatus status) {
+                RoleVO newRoleVO = roleDao.persist(new RoleVO(name, role.getRoleType(), description));
+                if (newRoleVO == null) {
+                    throw new CloudRuntimeException("Unable to add role into DB, is DB full?");
```

Comment:
nit - Can you improve the exception message?

Reply:
> nit - Can you improve the exception message?

Updated